### PR TITLE
Introduce a global backoff to rate limit failed image downloads

### PR DESCRIPTION
**Pull request by justinsb**
This should help with runaway downloads of protokube.tar.gz

### CODE REVIEW

File: pkg/backoff/global.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// The primary use is to rate-limit repeated downloads, to prevent runaway bandwidth bills
+func DoGlobalBackoff(err error) {
+	pause := computeBackoff()
+
+	glog.Warningf("inserting rate-limiting pause of %v after error: %v", pause, err)
+	time.Sleep(pause)
+}
+
+// computeBackoff computes the next backoff value, by doubling the backoff value, capping it at maxGlobalBackoff
+func computeBackoff() time.Duration {
```
Comment (gambol99) [inline pkg/backoff/global.go]: note https://github.com/kubernetes/kops/tree/master/vendor/github.com/jpillora/backoff is currently vendored ... we could swap for that .. or perhaps swap that for this at some point

Comment (justinsb) [reply pkg/backoff/global.go]: I do want to backport this, so I did something super-simple...

**Comment by mikesplain** [comment]
We could easily swap it out later but I like the idea of having something simple like this for now (and more backportable).

/lgtm

**Pull request merged**